oidc.go

```diff
@@ -101,8 +102,14 @@ func NewProvider(ctx context.Context, issuer string) (*Provider, error) {
 		return nil, fmt.Errorf("%s: %s", resp.Status, body)
 	}
 	defer resp.Body.Close()
+
 	var p providerJSON
 	if err := json.Unmarshal(body, &p); err != nil {
+		var isJSON bool
+		content, _, _ := mime.ParseMediaType(r.Header.Get("Content-Type"))
```

Comment:
Should we be smart about the err value here? e.g.

if err == nil && content != "application/json"? {